sdk/core/azure-core/src/main/java/com/azure/core/util/CoreUtils.java

```diff
@@ -376,4 +386,89 @@ public static Context mergeContexts(Context into, Context from) {
 
         return returnContext;
     }
+
+    /**
+     * Consumes response body if necessary. I.e. makes sure all bytes are read from network.
```

Comment:
>ry. I.e. makes sure all bytes are read from network. [](http://example.com/codeflow?start=40&length=52)

Curious why is it necessary to read all the bytes. Is the network stream shared between calls and so flushing the stream is necessary?

Reply:
the http clients are (usually) implemented in such a way that they'd terminate connection if there's pending data in either direction when request is "disposed/closed". That would prevent the connection from coming back to the pool and being reused.